source/slang/slang-ir-redundancy-removal.cpp

```diff
@@ -126,6 +126,197 @@ bool removeRedundancy(IRModule* module, bool hoistLoopInvariantInsts)
     return changed;
 }
 
+bool isAddressMutable(IRInst* inst)
+{
+    auto rootType = getRootAddr(inst)->getDataType();
+    switch (rootType->getOp())
+    {
+    case kIROp_ParameterBlockType:
+    case kIROp_ConstantBufferType:
+    case kIROp_StructuredBufferLoad:
```

Comment:
This is not correct. The only case that `kIROp_StructuredBufferLoad` can be an address root is in `StructuredBuffer<int*> buf`, and `let ptr = buf[0]`. In this case, the address `ptr` itself is actually mutable.